runtime/src/tiered_storage.rs

```diff
@@ -53,17 +64,73 @@ impl TieredStorage {
     pub fn path(&self) -> &Path {
         self.path.as_path()
     }
+
+    /// Writes the specified accounts into this TieredStorage.
+    pub fn append_accounts<
+        'a,
+        'b,
+        T: ReadableAccount + Sync,
+        U: StorableAccounts<'a, T>,
+        V: Borrow<Hash>,
+    >(
+        &self,
+        accounts: &StorableAccountsWithHashesAndWriteVersions<'a, 'b, T, U, V>,
+        skip: usize,
+    ) -> Option<Vec<StoredAccountInfo>> {
+        let result;
+
+        {
+            let writer = TieredStorageWriter::new(&self.path, self.format.as_ref().unwrap());
+            result = writer.append_accounts(accounts, skip);
+        }
+
+        result
```

Comment:
It's mainly because in the later PR, we want the writer to close the file and then we open the file as read-only.

Will do the simplified version in this PR.